articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,563 @@
+
+For example, the [`H`](xref:microsoft.quantum.intrinsic.h) and [`MResetZ`](xref:microsoft.quantum.measurement.mresetz) operations are found in the [`Microsoft.Quantum.Instrinsic`](xref:microsoft.quantum.intrinsic) and [`Microsoft.Quantum.Measurement`](xref:microsoft.quantum.measurement) namespaces (part of the [Q# Standard Libraries](xref:microsoft.quantum.qsharplibintro)).
+As such, they can always be called via their *full* names, `Microsoft.Quantum.Intrinsic.H(<qubit>)` and `Microsoft.Quantum.Measurement.MResetZ(<qubit>)`, but always doing this would lead to very cluttered code.
+
+Instead, `open` statements allow callables to be referenced with more concise shorthand, as we've done in the operation body above.
+The full Q# file containing our operation would therefore consist of defining our own namespace, opening the namespaces for those callables our operation uses, and then our operation:
+
+```qsharp
+namespace NamespaceName {
+    open Microsoft.Quantum.Intrinsic;     // for the H operation
+    open Microsoft.Quantum.Measurement;   // for MResetZ
+
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) { 
+            H(q);
+            return MResetZ(q);
+        }
+    }
+}
+```
+
+> [!NOTE]
+> Namespaces can also be *aliased* when opened, which can be helpful if callable/type names in two namespaces conflict.
+> For example, we could instead use `open Microsoft.Quantum.Instrinsic as NamespaceWithH;` above, and then call `H` via `NamespaceWithH.H(<qubit>)`.
+
+> [!NOTE]
+> One exception to all of this is the [`Microsoft.Quantum.Core`](xref:microsoft.quantum.core) namespace, which is always automatically opened.
+> Therefore, callables like [`Length`](xref:microsoft.quantum.core.length) can always be used directly.
+
+### Execution on target machines
+
+Now the general execution model of a Q# program becomes clear.
+
+<br/>
+<img src="./images/general_execution_model.png" alt="Q# program execution diagram" width="400">
+
+Firstly, the specific callable to be executed has access to any other callables and types defined in the same namespace.
+It also access those from any of the [Q# libraries](xref:microsoft.quantum.libraries), but those must be referenced either via their full name, or through the use of `open` statements described above.
+
+The callable itself is then executed on a *[target machine](xref:microsoft.quantum.machines)*.
+Such target machines can be actual quantum hardware (as currently in private preview on Azure Quantum), or the multiple simulators available as part of the QDK.
+For our purposes here, the most useful target machine is an instance of the [full-state simulator](xref:microsoft.quantum.machines.full-state-simulator), `QuantumSimulator`, which calculates the program's behavior as if it were being executed on a noise-free quantum computer.
```

Comment:
```suggestion
The callable itself is then executed on a *[target machine](xref:microsoft.quantum.machines)*.
Such target machines can be actual quantum hardware or the multiple simulators available as part of the QDK.
For our purposes here, the most useful target machine is an instance of the [full-state simulator](xref:microsoft.quantum.machines.full-state-simulator), `QuantumSimulator`, which calculates the program's behavior as if it were being executed on a noise-free quantum computer.
```
I would remove that since it'll probably be forgotten to update in the future. In case we want to mention Azure Quantum when is available we can go and add t.